g2core/canonical_machine.cpp

```diff
@@ -829,7 +829,11 @@ void cm_set_model_target(const float target[], const bool flags[], const cmUnits
     for (axis = AXIS_X; axis <= LAST_LINEAR_AXIS; axis++) {
         if (flags[axis] && (AXIS_STANDARD == cm->a[axis].axis_mode || AXIS_INHIBITED == cm->a[axis].axis_mode)) {
             if (ABSOLUTE_DISTANCE_MODE == cm->gm.distance_mode) {
-                cm->gm.target[axis] = cm_get_combined_offset(axis);
+                if (UNIT_CONVERSION_REQUIRED == mm_mode_status) {
```

Comment:
Here we check to see if unit conversion is needed and then don't do a conversion if is? and do a conversion in the else?